rpcs3/Emu/SysCalls/Modules/cellNetCtl.cpp

```diff
@@ -330,16 +341,77 @@ s32 cellNetCtlGetInfo(s32 code, vm::ptr<CellNetCtlInfo> info)
 		freeifaddrs(ifaddr);
 #endif
 	}
+	else
+	{
+		cellNetCtl.Todo("cellNetCtlGetInfo(code=0x%x (%s), info=*0x%x)", code, InfoCodeToName(code), info);
+	}
 
 	return CELL_OK;
 }
 
+void dialogOpenCallback()
+{
+	named_thread_t(WRAP_EXPR("SignInDialog Thread"), [=]()
+	{
+		while (g_sign_in_dialog->state == signInDialogOpen)
+		{
+			if (Emu.IsStopped())
+			{
+				g_sign_in_dialog->state = signInDialogAbort;
+				break;
+			}
+
+			std::this_thread::sleep_for(std::chrono::milliseconds(1)); // hack
+		}
+
+		CallAfter([]()
+		{
+			g_sign_in_dialog->Destroy();
+			g_sign_in_dialog->state = signInDialogNone;
+		});
+	}).detach();
+}
+
 s32 cellNetCtlNetStartDialogLoadAsync(vm::ptr<CellNetCtlNetStartDialogParam> param)
 {
 	cellNetCtl.Warning("cellNetCtlNetStartDialogLoadAsync(param=*0x%x)", param);
 
-	// TODO: Actually sign into PSN or an emulated network similar to PSN
-	sysutilSendSystemCommand(CELL_SYSUTIL_NET_CTL_NETSTART_FINISHED, 0);
+	// TODO: Actually sign into PSN or an emulated network similar to PSN (ESN)
+	sysutilSendSystemCommand(CELL_SYSUTIL_NET_CTL_NETSTART_LOADED, 0);
+
+	// The way this is handled, is heavily inspired by the cellMsgDialogOpen2 implementation
+	if (1 && param->type == CELL_NET_CTL_NETSTART_TYPE_NP) // Currently disabled until ESN support is in
```

Comment:
These changes are likely wrong. If you understood them, you wouldn't copypaste them from msgdialog anyway. Just remove this from cellNetCtl completely (you still have SignInDialog.cpp draft). These functions should be faked for now in the way game would think that there is no Internet, PSN connection or both, according to circumstances. It is the best way for now.